core/src/main/java/org/apache/carbondata/core/datamap/dev/DataMap.java

```diff
@@ -54,6 +56,13 @@ void init(DataMapModel dataMapModel)
   List<T> prune(Expression filter, SegmentProperties segmentProperties,
       List<PartitionSpec> partitions, CarbonTable carbonTable) throws IOException;
 
+  /**
+   * Prune the data maps for finding the row count. It returns a Map of
+   * blockletpath and the row count
+   */
+  Map<String, Long> getRowCount(Segment segment, List<PartitionSpec> partitions,
+      Map<String, Long> blockletToRowCountMap, boolean isUpdateFlow) throws IOException;
```

Comment:
I don't think datamap interface users need to know about update flow. Make it simplified, add another method to get all row count. So add two methods instead of one method for simplification.
```
getRowCountForEachBlock()
getRowCount()
```